lib/virt_autotest/utils.pm

```diff
@@ -116,6 +117,42 @@ sub is_kvm_host {
     return check_var("SYSTEM_ROLE", "kvm") || check_var("HOST_HYPERVISOR", "kvm") || check_var("REGRESSION", "qemu-hypervisor");
 }
 
+#retrun 1 if libvirt 9.0- is running which monolithic libvirtd is the default service
+sub is_monolithic_libvirtd {
+    unless (is_alp) {
+        return 1 if script_run('rpm -q libvirt-libs | grep -e "libs-9\.0" -e "libs-[1-8]\."') == 0;
+    }
+}
+
+# For legacy libvird, set debug level logging for libvirtd services
+# For modular libvirt, do the same settings to /etc/libvirt/virt{qemu,xen,driver}d.conf.
+# virt{qemu,xen}d daemons provide the most important libvirt log(sufficient for most issues).
+# virt{driver}.d daemons is only required by specific issues, eg virtual network failures may need virtnetworkd log.
+# But our automation is better to set them to collect more logs as we could as possible.
+# Developer asked to use different log file as log_output per daemon.
+sub turn_on_libvirt_debugging_log {
+
+    my @libvirt_daemons = is_monolithic_libvirtd ? "libvirtd" : qw(virtqemud virtstoraged virtnetworkd virtnodedevd virtsecretd virtproxyd virtlockd virtlogd);
+
+    #turn on debug and log filter for libvirt services
+    #set log_level = 1 'debug'
+    #the size of libvirtd with debug level and without any filter on sles15sp3 xen is over 100G,
+    #which consumes all the disk space. Now get comfirmation from virt developers,
+    #log filter is set to store component logs with different levels.
+    foreach my $daemon (@libvirt_daemons) {
+        my $conf_file = "/etc/libvirt/$daemon.conf";
+        if (script_run("ls $conf_file") == 0) {
+            script_run "sed -i '/^[# ]*log_level *=/{h;s/^[# ]*log_level *= *[0-9].*\$/log_level = 1/};\${x;/^\$/{s//log_level = 1/;H};x}' $conf_file";
+            script_run "sed -i '/^[# ]*log_outputs *=/{h;s%^[# ]*log_outputs *=.*[0-9].*\$%log_outputs = \"1:file:/var/log/libvirt/$daemon.log\"%};\${x;/^\$/{s%%log_outputs = \"1:file:/var/log/libvirt/$daemon.log\"%;H};x}' $conf_file";
+            script_run "sed -i '/^[# ]*log_filters *=/{h;s%^[# ]*log_filters *=.*[0-9].*\$%log_filters = \"1:qemu 1:libvirt 4:object 4:json 4:event 3:util 1:util.pci\"%};\${x;/^\$/{s%%log_filters = \"1:qemu 1:libvirt 4:object 4:json 4:event 3:util 1:util.pci\"%;H};x}' $conf_file";
+        }
+    }
+    script_run "grep -e 'log_level.*=' -e 'log_outputs.*=' -e 'log_filters.*=' /etc/libvirt/*d.conf";
+    save_screenshot;
+
+    restart_libvirtd if is_monolithic_libvirtd;
```

Comment:
For modular daemons, they also need restart.

Reply:
Is it realy required? In my test, "/var/log/virt*d.log" are created without restart any libvirt services as it is defined as `log_outputs = \"1:file:/var/log/libvirt/$daemon.log` in its configuration file.

Reply:
In https://www.libvirt.org/kbase/debuglogs.html#modular-vs-monolithic-daemons, it gives below. So I think it should be yes, just like monolithic libvirtd. But I can't explain why in your case no restart needed...

```
find & replace, or set the appropriate [Log outputs](https://www.libvirt.org/kbase/debuglogs.html#log-outputs) and [Log filters](https://www.libvirt.org/kbase/debuglogs.html#log-filters), e.g

log_filters="3:remote 4:event 3:util.json 3:rpc 1:*"
log_outputs="1:file:/var/log/libvirt/libvirtd.log"
save and exit

restart the corresponding service/daemon 
```

Reply:
I'll run a test to confirm whether these daemons are required to be restarted in TW.

Reply:
I'll run a test to confirm whether these daemons are required to be restarted in TW.

Reply:
Restarting all daemons in my test. It took 40s and seems not noisy.